orchestration/dagster_orchestration/hca_manage/manage.py

```diff
@@ -231,6 +231,32 @@ def submit_snapshot_request(self, qualifier: Optional[str] = None) -> str:
         logging.info(f"Snapshot creation job id: {response.id}")
         return response.id
 
+    def delete_snapshot(self, snapshot_name: Optional[str] = None, snapshot_id: Optional[str] = None):
+        if snapshot_name and not snapshot_id:
+            response = self.data_repo_client.enumerate_snapshots(filter=snapshot_name)
+            snapshot_id = response.items[0].id
+        elif snapshot_id and not snapshot_name:
+            pass  # let snapshot_id argument pass through
+        else:
+            # can't have both/neither provided
+            raise ValueError("You must provide either snapshot_name or snapshot_id, and cannot provide neither/both.")
+        response = self.data_repo_client.delete_snapshot(snapshot_id)
```

Comment:
🔧 Same comment about checking the response

Reply:
Just to document: The client already raises an exception with the status code and error details.